trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js

```diff
@@ -175,43 +157,99 @@ const submitDealAfterUkefIds = async (dealId, dealType, checker) => {
 
 exports.submitDealAfterUkefIds = submitDealAfterUkefIds;
 
-const submitDealPUT = async (req, res) => {
-  const { dealId, dealType, checker } = req.body;
-  let deal;
+/**
+ * Submits a deal to TFM before the UKEF IDs are generated.
+ * @param {string} dealId - The ID of the deal to be submitted.
+ * @param {string} dealType - The type of the deal.
+ * @param {string} checker - The name of the checker.
+ * @returns {Promise<Object>} - A promise that resolves to true if the submission is successful, false otherwise.
```

Comment:
Promise<object | boolean> I think? We get a object back from the api response if good and false if bad